packages/mobile/src/identity/privacy.ts

```
@@ -0,0 +1,138 @@
+async function getPhoneHashPrivate(e164Number: string): Promise<PhoneNumberHashDetails> {
+  const salt = await getPhoneNumberSalt(e164Number)
+  const phoneHash = getPhoneHash(e164Number, salt)
+  return {
+    e164Number,
+    phoneHash,
+    salt,
+  }
+}
+
+async function getPhoneNumberSalt(e164Number: string) {
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Getting phone number salt')
+
+  if (!isE164Number(e164Number)) {
+    throw new Error(ErrorMessages.INVALID_PHONE_NUMBER)
+  }
+
+  // TODO remove when iOS works
+  if (Platform.OS === 'ios') {
+    return 'fakeSalt'
+  }
+
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Retrieving blinded message')
+  const base64BlindedMessage = await BlindThresholdBls.blindMessage(e164Number)
+  const base64BlindSig = await postToSignMessage(base64BlindedMessage)
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Retrieving unblinded signature')
+  const base64UnblindedSig = await BlindThresholdBls.unblindMessage(base64BlindSig)
+  Logger.debug(`${TAG}@getPhoneNumberSalt`, 'Converting sig to salt')
+  return getSaltFromThresholdSignature(base64UnblindedSig)
+}
+
+interface SignMessageResponse {
+  success: boolean
+  // TODO change this in service
+  salt: string
+}
+
+// Send the blinded message off to the phone number privacy service and
+// get back the theshold signed blinded message
+async function postToSignMessage(base64BlindedMessage: string) {
+  Logger.debug(`${TAG}@postToSignMessage`, 'Retrieving blinded message')
+  const res = await fetch(PHONE_NUM_PRIVACY_SERVICE + SIGN_MESSAGE_ENDPOINT, {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      blindPhoneNumber: base64BlindedMessage,
+    }),
+  })
+
+  if (!res.ok) {
+    handleSignMessageFailure(res)
+  }
+
+  Logger.debug(`${TAG}@postToSignMessage`, 'Response ok. Parsing.')
+  const signResponse = (await res.json()) as SignMessageResponse
```

Do we want to `try ... catch` here?

---

I'd rather let it propagate and have the try/catch in `fetchPhoneHashPrivate` handle it